darts/models/torch_forecasting_model.py

```diff
@@ -622,74 +631,70 @@ def _prepare_predict_tensors(self, n: int, input_series_dataset: TimeSeriesInfer
 
         Returns
         -------
-        torch.Tensor of shape `(len(input_series_dataset), self.input_chunk_length, target.width + covariates.width)`
-            Initial input to model which contains the input target sereis and the matching past covariates,
-            if available
-        Optional[torch.Tensor] of shape `(len(input_series_dataset), n, covariates.width)`
+        Sequence[TimeSeries]
+            Past target series which are going to be predicted into the future, and which will used as
+            input to the model.
+        Optional[Sequence[TimeSeries]]
+            Covariates required to predict horizons up to `self.output_chunk_length`, or `None` if the model
+            does not require covariates.
+        Optional[Sequence[TimeSeries]]
             Covariates required to predict horizons beyond `self.output_chunk_length`, or `None` if the model
             does require future covariates either because it does not require covariates at all or because
             `n <= self.output_chunk_length`.
     """
 
-        in_past_arr = []
+        tgt_past_arr = []
+        cov_past_arr = []
         cov_future_arr = []
 
-        for target_series, covariate_series in input_series_dataset:
+        covariates = [None] * len(series) if covariates is None else covariates
+        raise_if_not(len(series) == len(covariates),
+                     'The number of target series must be equal to the number of covariates.')
+
+        for target_series, covariate_series in zip(series, covariates):
             raise_if_not(len(target_series) >= self.input_chunk_length,
-                            'All input series must have length >= `input_chunk_length` ({}).'.format(
+                         'All input series must have length >= `input_chunk_length` ({}).'.format(
                 self.input_chunk_length))
 
             # TODO: here we could be smart and handle cases where target and covariates do not have same time axis.
```

Comment:
I think we could perhaps remove this comment, as the logic you implemented does this (more or less)

Reply:
Done